contracts/EVMScriptFactories/UpdateTargetValidatorLimits.sol

```diff
@@ -0,0 +1,105 @@
+        bool isTargetLimitActive;
+        uint256 targetLimit;
+    }
+
+    // -------------
+    // ERRORS
+    // -------------
+
+    string private constant ERROR_NODE_OPERATOR_INDEX_OUT_OF_RANGE =
+        "NODE_OPERATOR_INDEX_OUT_OF_RANGE";
+    string private constant NODE_OPERATORS_IS_NOT_SORTED = "NODE_OPERATORS_IS_NOT_SORTED";
+
+    // -------------
+    // VARIABLES
+    // -------------
+
+    /// @notice Address of NodeOperatorsRegistry contract
+    INodeOperatorsRegistry public immutable nodeOperatorsRegistry;
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(
+        address _trustedCaller,
+        address _nodeOperatorsRegistry
+    ) TrustedCaller(_trustedCaller) {
+        nodeOperatorsRegistry = INodeOperatorsRegistry(_nodeOperatorsRegistry);
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    function createEVMScript(
+        address _creator,
+        bytes memory _evmScriptCallData
+    ) external view override onlyTrustedCaller(_creator) returns (bytes memory) {
+        TargetValidatorsLimit[] memory decodedCallData = abi.decode(
+            _evmScriptCallData,
+            (TargetValidatorsLimit[])
+        );
+
+        _validateInputData(decodedCallData);
+
+        bytes[] memory updateTargetLimitsCallData = new bytes[](decodedCallData.length);
+
+        for (uint256 i = 0; i < decodedCallData.length; i++) {
+            updateTargetLimitsCallData[i] = abi.encode(decodedCallData[i]);
+        }
+    
+        return
+            EVMScriptCreator.createEVMScript(
+                address(nodeOperatorsRegistry),
+                nodeOperatorsRegistry.updateTargetValidatorsLimits.selector,
+                updateTargetLimitsCallData
+            );
+    }
+
+    function decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    ) external pure returns (TargetValidatorsLimit[] memory) {
+        return _decodeEVMScriptCallData(_evmScriptCallData);
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    function _decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    ) private pure returns (TargetValidatorsLimit[] memory) {
+        return abi.decode(_evmScriptCallData, (TargetValidatorsLimit[]));
+    }
+
+    function _validateInputData(TargetValidatorsLimit[] memory _decodedCallData) private view {
+        uint256 nodeOperatorsCount = nodeOperatorsRegistry.getNodeOperatorsCount();
+        for (uint256 i = 0; i < _decodedCallData.length; i++) {
+            require(
+                i == 0 ||
+                    _decodedCallData[i].nodeOperatorId >
+                    _decodedCallData[i - 1].nodeOperatorId,
+                NODE_OPERATORS_IS_NOT_SORTED
+            );
+            require(
+                _decodedCallData[i].nodeOperatorId < nodeOperatorsCount,
+                ERROR_NODE_OPERATOR_INDEX_OUT_OF_RANGE
+            );
+        }
+    }
```

Comment:
The NOR contract also contains a check that `targetLimit <= UINT64_MAX`, pls consider adding this check here